backend/tests/lib.rs

```diff
@@ -16,43 +32,136 @@ fn test_connected() {
     assert_eq!(response.into_string().unwrap(), "Ok");
 }
 
-#[test]
-fn test_signup() {
+fn test_email() {
     let client = common::get_client().lock();
     let name: String = std::iter::repeat(())
         .map(|()| thread_rng().sample(Alphanumeric))
         .map(char::from)
         .take(16)
         .collect();
+    // set verification code: Request Time 1
+    let response = client
+        .post("/users/email")
+        .json(&json!({
+            "email": format!("{}@mails.tsinghua.edu.cn", name)
+        }))
+        .remote("127.0.0.1:8000".parse().unwrap())
+        .dispatch();
+    assert_eq!(response.status(), Status::Ok);
+    println!("{}", response.into_string().unwrap());
```

Comment:
use assert_eq to match the response.

Reply:
you mean without '!' ?

Reply:
No, change your println! to assert_eq! , to match the body of response.